common/membership/hashring.go

```diff
@@ -27,92 +27,104 @@ import (
 	"time"
 
 	"github.com/dgryski/go-farm"
-	"github.com/uber/ringpop-go/events"
 	"github.com/uber/ringpop-go/hashring"
-	"github.com/uber/ringpop-go/swim"
 
 	"github.com/uber/cadence/common"
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 	"github.com/uber/cadence/common/types"
 )
 
+//go:generate mockgen -package $GOPACKAGE -source $GOFILE -destination peerprovider_mock.go github.com/uber/cadence/common/membership PeerProvider
+
 // ErrInsufficientHosts is thrown when there are not enough hosts to serve the request
 var ErrInsufficientHosts = &types.InternalServiceError{Message: "Not enough hosts to serve the request"}
 
 const (
-	// RoleKey label is set by every single service as soon as it bootstraps its
-	// ringpop instance. The data for this key is the service name
-	RoleKey                = "serviceName"
 	minRefreshInternal     = time.Second * 4
 	defaultRefreshInterval = time.Second * 10
 	replicaPoints          = 100
 )
 
-type ringpopHashring struct {
-	status      int32
-	service     string
-	rp          *RingpopWrapper
-	refreshChan chan struct{}
-	shutdownCh  chan struct{}
-	shutdownWG  sync.WaitGroup
-	logger      log.Logger
+// PeerProvider is used to retrieve membership information from provider
+type PeerProvider interface {
+	common.Daemon
+
+	GetMembers(service string) ([]string, error)
+	WhoAmI() (string, error)
+	SelfEvict() error
+	Subscribe(name string, notifyChannel chan<- *ChangedEvent) error
+}
 
-	ringValue atomic.Value // this stores the current hashring
+// HostInfo is a type that contains the info about a cadence host
+type HostInfo struct {
+	addr string // ip:port
```

Comment:
Not really/not at this time

This string is used for hashing purposes, at this layer, it has nothing to do with actual addressing. 
Also, in history/matching we are using `yarpc.WithShardKey(<string>)` which is all we need today.

But this can be revisited later